### ISSUE TITLE
User provided device_type of 'cuda', but CUDA is not available.

### ISSUE THREAD

**Issue by KuroRS** [gazingstars123/Anima-Standalone-Trainer#38]
\venv\lib\site-packages\torch\amp\autocast_mode.py:266: UserWarning: User provided device_type of 'cuda', but CUDA is not available.
in the end it's show
accelerator device: cpu

but when I run test with
import torch
print(f"Is CUDA available: {torch.cuda.is_available()}")
print(f"PyTorch version: {torch.__version__}")
result
Is CUDA available: True
PyTorch version: 2.7.0+cu128

**Comment by gazingstars123** [comment]
Please first to make sure to set your venv location inside the global settings (with the gear icon). If it still doesn't , please try to run and reinstall the venv manually with all the required packages. 

Provide further information with your system, configs, training logs, etc. if it still doesn't work.

**Comment by KuroRS** [comment]
I removed all my python (I have multiple version of it), removed all environment relate to python, removed Anima Trainer and reinstall Python 3.10 and Anima Trainer again.
still same error, refuse to use GPU. (I have set venv location inside global settings)
so I see this error is from torch, so I try to uninstall all torch and reinstall with cu130 stable version.
now I got error from anima_model.py instead with the same kind of error.

D:\Anima-Standalone-Trainer\library\anima_models.py:300: UserWarning: CUDA is not available or torch_xla is imported. Disabling autocast.
  @torch.amp.autocast(device_type='cuda', dtype=torch.float32)
W0426 16:08:11.159000 17580 venv\Lib\site-packages\torch\utils\flop_counter.py:29] triton not found; flop counting will not work for triton kernels
2026-04-26 16:08:12 INFO     Loading settings from           train_util.py:4774
                             D:\Anima-Standalone-Trainer\tra                   
                             ining-ui\jobs\mat\_merged_confi                   
                             g.toml...                                         
                    WARNING  cache_latents_to_disk is        train_util.py:4448
                             enabled, so cache_latents is                      
                             also enabled /                                    
                             cache_latents_to_diskが有効なた                   
                             め、cache_latentsを有効にします                   
2026-04-26 16:08:12 INFO     Loading dataset config from   train_network.py:632
                             D:\Anima-Standalone-Trainer\t                     
                             raining-ui\jobs\mat\dataset.t                     
                             oml                                               
                    INFO     prepare images.                 train_util.py:2120
                    INFO     get image size from name of     train_util.py:2003
                             cache files                                       

  0%|          | 0/20 [00:00<?, ?it/s]
100%|██████████| 20/20 [00:00<?, ?it/s]
                    INFO     set image size from cache       train_util.py:2033
                             files: 0/20                                       
                    INFO     found directory                 train_util.py:2057
                             C:\Users\KuroRS\Desktop\Mato\Lo                   
                             ra_Train\10_Mato contains 20                      
                             image files                                       
                    INFO     Using 24 workers for caption    train_util.py:2066
                             reading                                           

read caption:   0%|          | 0/20 [00:00<?, ?it/s]
read caption: 100%|██████████| 20/20 [00:00<00:00, 2489.87it/s]
                    INFO     20 train images with repeats.   train_util.py:2166
                    INFO     0 reg images with repeats.      train_util.py:2170
                    WARNING  no regularization images /      train_util.py:2175
                             正則化画像が見つかりませんでし                    
                             た                                                
                    INFO     [Dataset 0]                     config_util.py:580
                               batch_size: 2                                   
                               resolution: (1024, 1024)                        
                               resize_interpolation: None                      
                               enable_bucket: False                            
                                                                               
                               [Subset 0 of Dataset 0]                         
                                 image_dir:                                    
                             "C:\Users\KuroRS\Desktop\Mato\L                   
                             ora_Train\10_Mato"                                
                                 image_count: 20                               
                                 num_repeats: 1                                
                                 shuffle_caption: False                        
                                 keep_tokens: 1                                
                                 caption_dropout_rate: 0.05                    
                                 caption_dropout_every_n_epo                   
                             chs: 0                                            
                                 caption_tag_dropout_rate: 0                   
                                 caption_prefix:                               
                                 caption_suffix: None                          
                                 color_aug: False                              
                                 flip_aug: False                               
                                 face_crop_aug_range: None                     
                                 random_crop: False                            
                                 token_warmup_min: 1,                          
                                 token_warmup_step: 0,                         
                                 alpha_mask: False                             
                                 resize_interpolation: None                    
                                 custom_attributes: {}                         
                                 is_reg: False                                 
                                 class_tokens: None                            
                                 caption_extension: .txt                       
                                                                               
                                                                               
                    INFO     [Prepare dataset 0]             config_util.py:592
                    INFO     loading image sizes.             train_util.py:995
                    INFO     Using 24 workers for image size  train_util.py:998
                             loading                                           

loading image sizes:   0%|          | 0/20 [00:00<?, ?it/s]
loading image sizes: 100%|██████████| 20/20 [00:00<00:00, 19991.92it/s]
                    INFO     prepare dataset                 train_util.py:1015
                    WARNING  cache_text_encoder_outpu anima_train_network.py:42
                             ts_to_disk is enabled,                            
                             so                                                
                             cache_text_encoder_outpu                          
                             ts is also enabled                                
                    INFO     Migrating subset caption anima_train_network.py:57
                             dropout rate (0.05) to                            
                             global level for Anima                            
                             strategy                                          
                    INFO     preparing accelerator         train_network.py:724
                    INFO     Loading Qwen3 text      anima_train_network.py:103
                             encoder...                                        
                    INFO     Loading Qwen3 text encoder from anima_utils.py:207
                             D:\ComfyUI_windows_portable\Com                   
                             fyUI\models\text_encoders\qwen_                   
                             3_06b_base.safetensors                            
2026-04-26 16:08:25 INFO     Loaded Qwen3 state dict: <All   anima_utils.py:244
                             keys matched successfully>                        
2026-04-26 16:08:26 INFO     Loaded Qwen3 text encoder.      anima_utils.py:252
                             Parameters: 596,049,920                           
                    INFO     Loading Anima DiT...    anima_train_network.py:120
                    INFO     Loading Anima DiT from           anima_utils.py:53
                             D:\ComfyUI_windows_portable\Comf                  
                             yUI\models\diffusion_models\anim                  
                             aOfficial_preview3Base.safetenso                  
                             rs                                                
                    INFO     DiT config: model_channels=2048, anima_utils.py:84
                             num_blocks=28, num_heads=16,                      
                             use_llm_adapter=True                              
2026-04-26 16:09:23 INFO     Loaded Anima DiT successfully.  anima_utils.py:115
                             Parameters: 2,091,068,928                         
2026-04-26 16:09:24 INFO     Loading Anima VAE...    anima_train_network.py:145
                    INFO     Loading Anima VAE from          anima_utils.py:126
                             D:\ComfyUI_windows_portable\Com                   
                             fyUI\models\vae\qwen_image_vae.                   
                             safetensors                                       
                    INFO     Loaded Anima VAE successfully.  anima_utils.py:159
                    INFO     [Dataset 0]                     train_util.py:2635
                    INFO     caching latents with caching    train_util.py:1118
                             strategy.                                         
                    INFO     caching latents...              train_util.py:1168
accelerator device: cpu

  0%|          | 0/20 [00:00<?, ?it/s]

**Comment by gazingstars123** [comment]
Try running these command
```
.\venv\Scripts\accelerate.exe env
.\venv\Scripts\python.exe -c "import os, torch; print('ACCELERATE_USE_CPU=', os.getenv('ACCELERATE_USE_CPU')); print(torch.cuda.is_available(), torch.cuda.device_count(), torch.__version__)"
```

You should expect to see something like:
```
ACCELERATE_USE_CPU= None
torch.cuda.is_available() = True
device_count = 1
torch = 2.7.1+cu128
```

if ACCELERATE_USE_CPU is set to true, you might need to locate and delete the accelerate config yaml. The default is at this location, `C:\Users\<YourUsername>\.cache\huggingface\accelerate\default_config.yaml`. It might also be set somewhere inside your venv (unlikely)

**Comment by KuroRS** [comment]
first command.
(venv) PS D:\Anima-Standalone-Trainer> .\venv\Scripts\accelerate.exe env

Copy-and-paste the text below in your GitHub issue

- `Accelerate` version: 1.6.0
- Platform: Windows-10-10.0.26100-SP0
- `accelerate` bash location: D:\Anima-Standalone-Trainer\venv\Scripts\accelerate.exe
- Python version: 3.10.11
- Numpy version: 2.2.6
- PyTorch version (GPU?): 2.11.0+cu130 (True)
- PyTorch XPU available: False
- PyTorch NPU available: False
- PyTorch MLU available: False
- PyTorch SDAA available: False
- PyTorch MUSA available: False
- System RAM: 31.77 GB
- GPU type: NVIDIA GeForce RTX 5060 Ti
- `Accelerate` default config:
        - compute_environment: LOCAL_MACHINE
        - distributed_type: NO
        - mixed_precision: bf16
        - use_cpu: False
        - debug: False
        - num_processes: 1
        - machine_rank: 0
        - num_machines: 1
        - gpu_ids: 1
        - rdzv_backend: static
        - same_network: True
        - main_training_function: main
        - enable_cpu_affinity: True
        - downcast_bf16: no
        - tpu_use_cluster: False
        - tpu_use_sudo: False
        - tpu_env: []

second command.
(venv) PS D:\Anima-Standalone-Trainer> .\venv\Scripts\python.exe -c "import os, torch; print('ACCELERATE_USE_CPU=', os.getenv('ACCELERATE_USE_CPU')); print(torch.cuda.is_available(), torch.cuda.device_count(), torch.__version__)"
ACCELERATE_USE_CPU= None
True 1 2.11.0+cu130

**Comment by gazingstars123** [comment]
do you have something like this at this location? `C:\Users\<YourUsername>\.cache\huggingface\accelerate\default_config.yaml`. delete that if exists. 

Your issue is currently being `gpu_ids: 1`, when it should be `0`

**Comment by KuroRS** [comment]
> do you have something like this at this location? `C:\Users\<YourUsername>\.cache\huggingface\accelerate\default_config.yaml`. delete that if exists.
> 
> Your issue is currently being `gpu_ids: 1`, when it should be `0`

Thanks you my jesus, you saved me!
I don't know how that happen but it's using my GPU now.
